### PR TITLE
Added setting for python-specific connection URL 

### PR DESCRIPTION
**Pull request by Floris-Hyve**
### Problem:

Standard database connection string does not work when using MySQL python connector for migration. For example, ssl connection is disabled for the standard connection URL, but ssl connection is required for python MySQL connection. 

### Fix: 

Added Python-specific connection URL, and implemented it in the get_database_properties function, so the correct URL is selected for the migration script.

Input URL is given in the portal.properties:

`db.connection_string_python=`

### CODE REVIEW

File: docs/deployment/customization/portal.properties-Reference.md [hunk 1]
```diff
@@ -31,6 +31,22 @@ the `db.connection_string` instead.
...
 ```
 
+### Python-specific connection URL
+
+``` 
+db.connection_string_python= 
+
+``` 
+
+Setting for python-specific connection URL in order to connect using both python and java at the entrypoint.
```
Comment (pvannierop) [inline docs/deployment/customization/portal.properties-Reference.md:41]:
```suggestion
Setting that allow a python-specific database connection URL that is different from the connection URL used by Java. This allows, for instance, to migrate the database with different connection properties in the docker entry point script.
```

File: docs/deployment/customization/portal.properties-Reference.md [hunk 2]
```diff
@@ -31,6 +31,22 @@ the `db.connection_string` instead.
...
+
+The format of the `db.connection_string_python` is: 
+
+``` 
+jdbc:mysql://localhost:3306/cbiodb
+
+```
```
Comment (pvannierop) [inline docs/deployment/customization/portal.properties-Reference.md:48]:
```suggestion
The format of the `db.connection_string_python` is: 

``` 
jdbc:mysql://localhost:3306/cbiodb

``` 
```

The format of the `db.connection_string_python` is identical to `db.connection_string`.

File: src/main/resources/portal.properties.EXAMPLE [hunk 3]
```diff
@@ -6,6 +6,8 @@ db.user=cbio_user
 db.password=somepassword
 db.driver=com.mysql.jdbc.Driver
 db.connection_string=jdbc:mysql://localhost:3306/cbioportal?useSSL=false
+# Added connection string for when both python and java connections are used in the entrypoint
```
Comment (pvannierop) [inline src/main/resources/portal.properties.EXAMPLE:9]: Please better describe what this setting __does__, not only when it is used.

**Review by pvannierop — CHANGES_REQUESTED** [review]
Some small changes.

File: src/main/resources/portal.properties.EXAMPLE [hunk 4]
```diff
@@ -6,6 +6,8 @@ db.user=cbio_user
 db.password=somepassword
 db.driver=com.mysql.jdbc.Driver
 db.connection_string=jdbc:mysql://localhost:3306/cbioportal?useSSL=false
+# Connection string for when both python and java connections are used in the docker entrypoint. When a connection URL is given for python, it will be used to form a connection to the mySQL database instead of the standard connection string. This allows for different connection properties to be used (like ssl connection for example). 
+# db.connection_string_python= jdbc:mysql://cbioDB:3306/cbioportal
```
Comment (pvannierop) [inline src/main/resources/portal.properties.EXAMPLE:10]: The comment on precedence should also  be made in the portal.properties-reference.md

**Review by pvannierop — CHANGES_REQUESTED** [review]
Two proposed changes:
1. Clean up the commits (now there are 7!)
2. Harmonize information on precedence of the URL properties.

**Comment by JREastonMarks** [comment]
From an RFC72 (6.0) perspective the core module has been moved out of the cbioportal code base and into its own code base here: https://github.com/cBioPortal/cbioportal-core

@pvannierop can you create another PR for this in that repo? That way it will work in both 5.X and 6.X versions of cBioPortal.

**Comment by alisman** [comment]
@JREastonMarks can you provide direction on how re-target this PR in a 6.0 world.

**Comment by JREastonMarks** [comment]
Hi @Floris-Hyve,
With the release of 6.0.0 (RFC72) we moved a lot of code around and updated some of the files that you reference in this PR. The core module has been moved to its own repository: https://github.com/cBioPortal/cbioportal-core. If you move this PR to the core package it should still be a quick change. Please let me know if you need any help.

**Comment by JREastonMarks** [comment]
Closing but willing to reopen if needed.

**Pull request closed without merging**